api/v1beta1/hypershiftserviceconfig_types.go

```diff
@@ -0,0 +1,69 @@
+/*
+Copyright 2021.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	conditionsv1 "github.com/openshift/custom-resource-status/conditions/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// HypershiftServiceConfigSpec defines the desired state of HypershiftServiceConfig.
+type HypershiftServiceConfigSpec struct {
+	AgentServiceConfigSpec `json:",inline"`
+
+	// KubeconfigSecretRef is the reference to the Secret containing a kubeconfig of an external
+	// control plane (e.g. hypershift HostedCluster). When specified, the assisted-service
+	// deployment would use that kubeconfig instead of the in-cluster used by default.
+	// Using this scheme, assisted-service controllers watch and reconcile entities on the HostedCluster
```

Comment:
Not sure `scheme` is the right word here.
```suggestion
	// In this way, assisted-service controllers will watch and reconcile entities on the HostedCluster
```

Reply:
text deleted completely and replaced by a one liner